sdk/core/Azure.Core.Experimental/src/JsonData.cs

```diff
@@ -0,0 +1,1090 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
@pakrym The commit message makes this a little more clear but I was worried about hacking and slashing the existing `DynamicJson` type when we really have no idea what shape this is going to end up in.  The long term goal would be that `JsonData` replaces `DynamicJson`, but I wasn't sure what other components where depending on it right now and didn't want to take the pain of trying to push through API changes in unrelated areas while we drive the design and experiment.

I copied the tests and added a few additional ones.

Let me know if there's something else you'd prefer here, as I get this is a little wacky.

Reply:
Very few (if any) components are using the DynamicJson right now. I would encourage you to try and evolve the `DynamicJson` if possible. If it becomes too much pain I'm OK making a copy.

Reply:
Ended up just removing `DynamicJson` and moved everything over to `JsonData`.  As you suspected it was not that hard.